backend/settings.py

```diff
@@ -0,0 +1,662 @@
+    allow_partial_result: bool = False
+    include_contexts: Optional[List[str]] = ["citations", "intent"]
+    vectorization_dimensions: Optional[int] = None
+    role_information: str = Field(
+        validation_alias="AZURE_OPENAI_SYSTEM_MESSAGE"
+    )
+
+    @field_validator('include_contexts', mode='before')
+    @classmethod
+    def split_contexts(cls, comma_separated_string: str, info: ValidationInfo) -> List[str]:
+        if isinstance(comma_separated_string, str) and len(comma_separated_string) > 0:
+            return parse_multi_columns(comma_separated_string)
+        
+        return cls.model_fields[info.field_name].get_default()
+
+
+class DatasourcePayloadConstructor(BaseModel, ABC):
+    _settings: '_AppSettings' = PrivateAttr()
+    
+    def __init__(self, settings: '_AppSettings', **data):
+        super().__init__(**data)
+        self._settings = settings
+    
+    @abstractmethod
+    def construct_payload_configuration(
+        self,
+        *args,
+        **kwargs
+    ):
+        pass
+
+
+class _AzureSearchSettings(BaseSettings, DatasourcePayloadConstructor):
+    model_config = SettingsConfigDict(
+        env_prefix="AZURE_SEARCH_",
+        env_file=DOTENV_PATH,
+        extra="ignore",
+        env_ignore_empty=True
+    )
+    service: str = Field(exclude=True)
+    index: str = Field(serialization_alias="index_name")
+    key: Optional[str] = Field(default=None, exclude=True)
+    use_semantic_search: bool = Field(default=False, exclude=True)
+    semantic_search_config: str = Field(default="", serialization_alias="semantic_configuration")
+    content_columns: Optional[List[str]] = Field(default=None, exclude=True)
+    vector_columns: Optional[List[str]] = Field(default=None, exclude=True)
+    title_column: Optional[str] = Field(default=None, exclude=True)
+    url_column: Optional[str] = Field(default=None, exclude=True)
+    filename_column: Optional[str] = Field(default=None, exclude=True)
+    query_type: Literal[
+        'simple',
+        'vector',
+        'semantic',
+        'vector_simple_hybrid',
+        'vector_semantic_hybrid'
+    ] = "simple"
+    permitted_groups_column: Optional[str] = Field(default=None, exclude=True)
+    
+    # Constructed fields
+    endpoint: Optional[str] = None
+    authentication: Optional[dict] = None
+    embedding_dependency: Optional[dict] = None
+    fields_mapping: Optional[dict] = None
+    filter: Optional[str] = None
+    
+    @field_validator('content_columns', 'vector_columns', mode="before")
+    @classmethod
+    def split_columns(cls, comma_separated_string: str) -> List[str]:
+        if isinstance(comma_separated_string, str) and len(comma_separated_string) > 0:
+            return parse_multi_columns(comma_separated_string)
+        
+        return None
+    
+    @model_validator(mode="after")
+    def set_endpoint(self) -> Self:
+        self.endpoint = f"https://{self.service}.search.windows.net"
```

Comment:
move ".search.windows.net" to some settings? so we can change it easily for gov cloud

Reply:
Done!